swarm/network/kademlia.go

```diff
@@ -53,14 +53,15 @@ var pof = pot.DefaultPof(256)
 // KadParams holds the config params for Kademlia
 type KadParams struct {
 	// adjustable parameters
-	MaxProxDisplay int // number of rows the table shows
-	MinProxBinSize int // nearest neighbour core minimum cardinality
-	MinBinSize     int // minimum number of peers in a row
-	MaxBinSize     int // maximum number of peers in a row before pruning
-	RetryInterval  int // initial interval before a peer is first redialed
-	RetryExponent  int // exponent to multiply retry intervals with
-	MaxRetries     int // maximum number of redial attempts
-	PruneInterval  int // interval between peer pruning cycles
+	MaxProxDisplay int  // number of rows the table shows
+	MinProxBinSize int  // nearest neighbour core minimum cardinality
+	MinBinSize     int  // minimum number of peers in a row
+	MaxBinSize     int  // maximum number of peers in a row before pruning
+	RetryInterval  int  // initial interval before a peer is first redialed
+	RetryExponent  int  // exponent to multiply retry intervals with
+	MaxRetries     int  // maximum number of redial attempts
+	PruneInterval  int  // interval between peer pruning cycles
+	HealthCheck    bool // whether kademlia On should signal completion
```

Comment:
exact health can only be calculated if you know all the peers so this is only for tests